common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationSupportSuite.java

```diff
@@ -2329,6 +2329,19 @@ public void testStringLocate() throws SparkException {
     assertStringLocate("b", "a🙃x🙃b", 4, "UTF8_LCASE", 5);
     assertStringLocate("b", "a🙃x🙃b", 4, "UNICODE", 5);
     assertStringLocate("b", "a🙃x🙃b", 4, "UNICODE_CI", 5);
+    // Out of bounds test cases.
+    assertStringLocate("a", "asd", 4, "UTF8_BINARY", 0);
+    assertStringLocate("a", "asd", 4, "UTF8_LCASE", 0);
+    assertStringLocate("a", "asd", 4, "UNICODE", 0);
+    assertStringLocate("a", "asd", 4, "UNICODE_CI", 0);
```

Comment:
```suggestion
    assertStringLocate("a", "asd", 4, "UTF8_BINARY", 0);
    assertStringLocate("a", "asd", 4, "UTF8_LCASE", 0);
    assertStringLocate("a", "asd", 4, "UNICODE", 0);
    assertStringLocate("a", "asd", 4, "UNICODE_CI", 0);
    assertStringLocate("", "asd", 100, "UTF8_BINARY", 100);
    assertStringLocate("", "asd", 100, "UTF8_LCASE", 100);
    assertStringLocate("", "asd", 100, "UNICODE", 100);
    assertStringLocate("", "asd", 100, "UNICODE_CI", 100);
```

Reply:
Also the suggested edit here is wrong, the expected result should be 1 (as per the UTF8 Binary collation).

Reply:
thanks, sorry - yes, "1" is what I meant